NitroxPatcher/Patches/Dynamic/OxygenManager_RemoveOxygen_Patch.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Reflection;
+using HarmonyLib;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.Helper;
+
+/// <summary>
+/// Prevents Local Player from loosing oxygen while he hasn't loaded yet
+/// Oxygen removal starts after PlayerPositionInitialSyncProcessor. If player is moved in the water because before, he'll still be in the spawn pod
+/// </summary>
+namespace NitroxPatcher.Patches.Dynamic
+{
+    public class OxygenManager_RemoveOxygen_Patch : NitroxPatch, IDynamicPatch
```

Comment:
Not a huge fan of those patches that will trigger at each frame.

I think we can use the same system as precursor teleporter which freeze your stats when you're teleporting and waiting for world settling.
If you look at Player::Update() : 
```csharp
	if (!this.CanBreathe() && !this.IsFrozenStats())
	{
		float breathPeriod = this.GetBreathPeriod();
		if (this.timeMonitor.DidChangeInterval(breathPeriod, 0f))
		{
			this.oxygenMgr.RemoveOxygen(this.GetOxygenPerBreath(breathPeriod, this.depthClass.value));
			this.tookBreathEvent.Trigger(this);
		}
	}
```
Might need to check but you can play with Player::FreezeStats() and Player::UnfreezeStats() to do the same thing (and also target all player stats, not only oxygen)

Reply:
That's really my bad for not seing it since I litteraly took inspiration from that code